Assets/Scripts/Hierarchical/HierarchicalBase.cs

```diff
@@ -54,6 +56,22 @@ public void ClearSubHierarchicals() {
     _subHierarchicals.Clear();
   }
 
+  public List<IHierarchical> LeafHierarchicals(bool activeOnly, bool withTargetOnly) {
+    var subHierarchicals = (activeOnly ? ActiveSubHierarchicals : SubHierarchicals).ToList();
+    if (subHierarchicals.Count > 0) {
+      var leafHierarchicals = new List<IHierarchical>();
+      foreach (var subHierarchical in subHierarchicals) {
+        leafHierarchicals.AddRange(subHierarchical.LeafHierarchicals(activeOnly, withTargetOnly));
+      }
+      return leafHierarchicals;
+    }
+
+    if (withTargetOnly && (Target == null || Target.IsTerminated)) {
+      return new List<IHierarchical>();
+    }
+    return new List<IHierarchical> { this };
```

Comment:
When called with `withTargetOnly: false`, the method returns leaf hierarchicals regardless of whether they have a target. but in `InterceptorBase.FixedUpdate()`:
```csharp
List<IHierarchical> leafHierarchicals =
    HierarchicalAgent.LeafHierarchicals(activeOnly: false, withTargetOnly: false);
NumSubInterceptorsPlannedRemaining = NumSubInterceptors - leafHierarchicals.Count;
```

This counts ALL leaf hierarchicals, including those that were created but never assigned targets. This would incorrectly exhaust `NumSubInterceptorsPlannedRemaining` too early, no?